packages/evm/test/registry/guardians-registry-locking.js

```diff
@@ -0,0 +1,265 @@
+        assertEvent(receipt, REGISTRY_EVENTS.LOCK_MANAGER_CHANGED, { expectedArgs: { lockManager: manager, allowed: true } })
+      })
+    }
+
+    const itCreatesTheActivationLock = () => {
+      it('creates the lock', async () => {
+        await registry.lockActivation(lockManager.address, lockAmount, { from: guardian })
+
+        const { amount, total } = await registry.getActivationLock(guardian, lockManager.address)
+        assertBn(amount, lockAmount, 'locked amount does not match')
+        assertBn(total, lockAmount, 'total locked amount does not match')
+      })
+
+      it('emits an event', async () => {
+        await registry.lockActivation(lockManager.address, lockAmount, { from: guardian })
+        const receipt = await registry.lockActivation(lockManager.address, lockAmount, { from: guardian })
+
+        assertAmountOfEvents(receipt, REGISTRY_EVENTS.GUARDIAN_ACTIVATION_LOCK_INCREASED)
+        assertEvent(receipt, REGISTRY_EVENTS.GUARDIAN_ACTIVATION_LOCK_INCREASED, { expectedArgs: { guardian, lockManager, amount: lockAmount.mul(bn(2)), total: lockAmount.mul(bn(2)) } })
+      })
+
+      it('can creates multiple locks', async () => {
+        await registry.lockActivation(lockManager.address, lockAmount, { from: guardian })
+        await registry.lockActivation(lockManager.address, lockAmount, { from: guardian })
+
+        await registry.changeLockManager(anotherLockManager.address, true, { from: governor })
+        await registry.lockActivation(anotherLockManager.address, lockAmount, { from: guardian })
+
+        const { amount, total } = await registry.getActivationLock(guardian, lockManager.address)
+        assertBn(amount, lockAmount.mul(bn(2)), 'locked amount does not match')
+        assertBn(total, lockAmount.mul(bn(3)), 'total locked amount does not match')
+      })
+
+      it('does not allow to deactivate the locked amount for present active tokens', async () => {
+        await activateTokens(lockAmount)
+
+        await registry.lockActivation(lockManager.address, lockAmount, { from: guardian })
+
+        await assertRevert(registry.deactivate(lockAmount, { from: guardian }), REGISTRY_ERRORS.DEACTIVATION_AMOUNT_EXCEEDS_LOCK)
+      })
+
+      it('does not allow to deactivate the locked amount for future active tokens', async () => {
+        await registry.lockActivation(lockManager.address, lockAmount, { from: guardian })
+
+        await activateTokens(lockAmount)
+
+        await assertRevert(registry.deactivate(lockAmount, { from: guardian }), REGISTRY_ERRORS.DEACTIVATION_AMOUNT_EXCEEDS_LOCK)
+      })
+    }
+
+    context('when the given lock manager is allowed', async () => {
+      allowLockManager()
+
+      context('when any lock manager is allowed', async () => {
+        allowLockManager(true)
+        itCreatesTheActivationLock()
+      })
+
+      context('when any lock manager is not allowed', async () => {
+        itCreatesTheActivationLock()
+      })
```

Comment:
Might be worth adding a quick note explaining the situation in the two cases above; was initially a bit confused on the additional use of (or lack of) `allowLockManager()`.